rust/serving/src/app/callback/store/redisstore.rs

```diff
@@ -109,8 +109,90 @@ async fn handle_write_requests(
 // It is possible to move the methods defined here to be methods on the Redis actor and communicate through channels.
 // With that, all public APIs defined on RedisConnection can be on &self (immutable).
 impl super::Store for RedisConnection {
+    async fn register(&mut self, id: Option<String>) -> StoreResult<String> {
+        match id {
+            Some(id) => {
+                let mut pipe = redis::pipe();
+                let key = format!("request:{id}:status");
+                pipe.cmd("SET").arg(&key).arg(STATUS_PROCESSING).arg("NX");
+                // if the ttl is configured, add the EXPIRE command to the pipeline
+                if let Some(ttl) = self.config.ttl_secs {
+                    pipe.arg("EX").arg(ttl);
+                }
+
+                let (status,): (bool,) =
+                    pipe.query_async(&mut self.conn_manager)
+                        .await
+                        .map_err(|e| {
+                            StoreError::StoreWrite(format!(
+                                "Registering request_id={id} in Redis: {e:?}"
+                            ))
+                        })?;
+
+                if !status {
+                    // The user specified request id already exists
+                    return Err(StoreError::DuplicateRequest(id));
+                }
+                Ok(id)
+            }
+            None => {
```

Comment:
if ID is generated from above, we do not have to do this. This will be an over complication once we have other stores (IPS, DDB, etc.)